sdk/src/utils/xmp_inmemory_utils.rs

```diff
@@ -130,13 +149,14 @@ fn add_xmp_key(xmp: &str, key: &str, value: &str) -> Result<String> {
                 if !added {
                     // didn't exist, so add it
                     elem.push_attribute((key, value));
+                    added = true;
```

Comment:
The lack of `added = true` is why the remote manifests url was being added multiple times.

Reply:
Thanks for tracking this down!